src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,199 @@
+package duke;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+public class TaskList {
+    /**
+     * Marks a task as done.
+     * 
+     * 
+     * @param input
+     * @throws DukeException
+     */
+    public static void mark(String input) throws DukeException {
+        ArrayList<String> arr = new ArrayList<>(Arrays.asList(input.split(" ")));
+        if (arr.get(1) == "") {
+            throw new DukeException(
+                    "Oops! Which task would you like to mark?"
+            );
+        }
+        int i = Integer.parseInt(arr.get(1));
+        if (i <= Duke.list.size() && i > 0) {
+            Task toBeMarked = Duke.list.get(Integer.parseInt(input.substring(5)) - 1);
+            toBeMarked.markAsDone();
+        } else {
+            throw new DukeException(
+                    "That is not a valid task number!"
+            );
+        }
+    }
+
+    /**
+     * Unmarks a task.
+     * 
+     * 
+     * @param input
+     * @throws DukeException
+     */
+    public static void unmark(String input) throws DukeException {
+        ArrayList<String> arr = new ArrayList<>(Arrays.asList(input.split(" ")));
+        if (arr.get(1) == "") {
+            throw new DukeException(
+                    "Oops! Which task would you like to unmark?"
+            );
+        }
+        int i = Integer.parseInt(arr.get(1));
+        if (i <= Duke.list.size() && i > 0) {
+            Task toBeMarked = Duke.list.get(Integer.parseInt(input.substring(7)) - 1);
+            toBeMarked.markAsUndone();
+        } else {
+            throw new DukeException(
+                    "That is not a valid task number!"
+            );
+        }
+    }
+
+    protected static void deadline(String input) throws DukeException {
+        if (stripDescription(input)[0] == "") {
+            throw new DukeException(
+                    "Oops, the description of deadline cannot be empty!"
+            );
+        }
+
+        if (stripDescription(input)[1] == "") {
+            throw new DukeException(
```

Comment:
Perhaps it may be better to do:
throw new DukeException
("Oops, please tell me when is this due!");
As per the java standards